dashboard/app/assets/javascripts/editor/json_editor.js

```diff
@@ -0,0 +1,106 @@
+/* global CodeMirror */
+
+function calloutEditor(container, options) {
+
```

Comment:
For JS assets like these in dashboard, they end up being concatenated and loaded as a single script. This means that top level functions we declare here are loaded into the global namespace. Obviously this is something we depend on in this case, but I like to make it explicit that this is what's happening, i.e. I would do something like:
```
window.calloutEditor = function (container, options) {
  ...
}
```
The other thing that can happen is that you miss a var and accidentally add a variable to the global namespace, for example
```
window.myfunc = function () {
  bar = 1; // suddenly window.bar = 1
}
```
You can protect against this by wrapping this all in an anonymous function, so you'd have something like:
```
(function () {
  window.calloutEditor = function (container, options) {
    ...
  }
})()